### PR TITLE
Add LLVM-C.dll on Windows

### LINKED ISSUE #212: Add LLVM-C.dll to build
### Comment:

https://github.com/conda-forge/llvmdev-feedstock/blob/2977f18493a7062562036fe588e021dac22a0da6/recipe/bld.bat#LL27C6-L27C30

LLVM-C.dll is disabled on Windows and I think it should be enabled to make developing with llvm-c API easier on Windows.

### PR DESCRIPTION
**Pull request by metab0t**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #212

### CODE REVIEW

**Comment by h-vetinari** [comment]
@isuruf, I'd very much like your opinion here. The first point is whether we should set `LLVM_BUILD_LLVM_C_DYLIB=ON` on osx as well (where, contrary to windows, it's off by default).

I see you added the `=NO` on windows in https://github.com/conda-forge/llvmdev-feedstock/commit/8059f3a076bfc1770ab993c22e882fc7279e66a3 (though I don't see relevant comments in the corresponding [PR](https://github.com/conda-forge/llvmdev-feedstock/pull/66)). That commit is also relevant because it introduced another point about this recipe I'm not 100% sure about - currently on windows, `libllvm16` & `llvm` are completely empty packages - I presume because this would be for the shared libraries which we're not building currently.

I see from the build log that `Library/bin/LLVM-C.dll` is shared though, so I wonder in which output this should go[^1]. The unix version of the library will end up in `libllvm16` AFAICT. Another, larger question is then whether we should try again to build the shared libraries on windows and populate `libllvm16` also on windows.

[^1]: that said, `llvmdev` currently contains [two](https://github.com/regro/libcfgraph/blob/master/artifacts/llvmdev/conda-forge/win-64/llvmdev-16.0.3-h2c25772_1.json#L299-L300) DLLs on windows already, `LTO.dll` & `Remarks.dll`

**Comment by h-vetinari** [comment]
PS. Thanks for the PR @metab0t! Sorry if this will still need some hashing out.

**Comment by h-vetinari** [comment]
Ping @isuruf. Might be good to figure this one out for the 17 cycle?

**Comment by isuruf** [comment]
This should be a separate output. `libllvm-c16`

**Comment by h-vetinari** [comment]
@isuruf, this is ready for review, PTAL. I refactored the installation scripts away from:
> install then delete unwanted bits

towards
> install into temporary prefix and copy specific files

which I think is clearer in intent and less error-prone, see https://github.com/conda-forge/llvmdev-feedstock/pull/213/commits/22fb387d61a1de9a5097d4f570b1bbb405ecaede.

This would be an independent improvement from this PR, but it makes sense in this context, because that way, it is a lot easier to extend for adding a libllvm-c output.

**Comment by h-vetinari** [comment]
Ping @isuruf if you want to review. Otherwise this follows your request of putting things in a separate output and I'll merge it in a few days.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -155,6 +157,36 @@ outputs:
...
 
+  # Contains LLVM-C shared library
+  - name: libllvm-c{{ major_ver }}
+    script: install_llvm.sh   # [unix]
+    script: install_llvm.bat  # [win]
+    build:
+      activate_in_script: true
+      # not supported on linux, see
+      # https://github.com/llvm/llvm-project/blob/llvmorg-16.0.6/llvm/tools/llvm-shlib/CMakeLists.txt#L82-L85
+      # TODO osx currently (16.0.6) does not build: raise issue as soon as we've caught up to last release
```
Comment (isuruf) [inline recipe/meta.yaml:168]: Add a run_exports?

Comment (h-vetinari) [reply recipe/meta.yaml:170]: Good point, thanks. Done

**Pull request merged**